packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -152,10 +160,85 @@ export class Subscription extends Resource {
+        }
+        const maxDelayTargetSecs = maxDelayTarget.toSeconds();
+        if (maxDelayTargetSecs < 1 || maxDelayTargetSecs > delayTargetLimitSecs) {
+          throw new Error(`maxDelayTarget must be between 1 and ${delayTargetLimitSecs} seconds inclusive, got: ${maxDelayTargetSecs}s`);
+        }
+        if ((minDelayTarget !== undefined) && minDelayTarget.toSeconds() > maxDelayTargetSecs) {
+          throw new Error('minDelayTarget must not exceed maxDelayTarget');
+        }
+      }
+
+      const numRetriesLimit = 100;
+      if (healthyRetryPolicy.numRetries && (healthyRetryPolicy.numRetries < 0 || healthyRetryPolicy.numRetries > numRetriesLimit)) {
+        throw new Error(`numRetries must be between 0 and ${numRetriesLimit} inclusive, got: ${healthyRetryPolicy.numRetries}`);
+      }
+      const { numNoDelayRetries, numMinDelayRetries, numMaxDelayRetries } = healthyRetryPolicy;
+      if (numNoDelayRetries && (numNoDelayRetries < 0 || !Number.isInteger(numNoDelayRetries))) {
+        throw new Error(`numNoDelayRetries must be an integer zero or greater, got: ${numNoDelayRetries}`);
+      }
+      if (numMinDelayRetries && (numMinDelayRetries < 0 || !Number.isInteger(numMinDelayRetries))) {
+        throw new Error(`numMinDelayRetries must be an integer zero or greater, got: ${numMinDelayRetries}`);
+      }
+      if (numMaxDelayRetries && (numMaxDelayRetries < 0 || !Number.isInteger(numMaxDelayRetries))) {
+        throw new Error(`numMaxDelayRetries must be an integer zero or greater, got: ${numMaxDelayRetries}`);
+      }
+    }
+    if (throttlePolicy) {
+      const maxReceivesPerSecond = throttlePolicy.maxReceivesPerSecond;
+      if (maxReceivesPerSecond !== undefined && (maxReceivesPerSecond < 1 || !Number.isInteger(maxReceivesPerSecond))) {
+        throw new Error(`maxReceivesPerSecond must be an integer greater than zero, got: ${maxReceivesPerSecond}`);
+      }
+    }
+    return {
+      healthyRetryPolicy: healthyRetryPolicy ? {
+        // minDelayTarget, maxDelayTarget and numRetries are (empirically) mandatory when healthyRetryPolicy is specified,
+        // but for user-friendliness we allow them to be undefined and set them here instead.
+        // The defaults we use here are the same used in the event healthyRetryPolicy is not specified, see https://docs.aws.amazon.com/sns/latest/dg/creating-delivery-policy.html.
+        minDelayTarget: (healthyRetryPolicy.minDelayTarget === undefined) ? 20 : healthyRetryPolicy.minDelayTarget.toSeconds(),
+        maxDelayTarget: (healthyRetryPolicy.maxDelayTarget === undefined) ? 20 : healthyRetryPolicy.maxDelayTarget.toSeconds(),
+        numRetries: (healthyRetryPolicy.numRetries === undefined) ? 3: healthyRetryPolicy.numRetries,
+        numNoDelayRetries: healthyRetryPolicy.numNoDelayRetries,
+        numMinDelayRetries: healthyRetryPolicy.numMinDelayRetries,
+        numMaxDelayRetries: healthyRetryPolicy.numMaxDelayRetries,
```

Comment:
Should these have default values?

Reply:
Similar to https://github.com/aws/aws-cdk/pull/30830#discussion_r1796169589 question, I guess it would depend on your view on that :sweat_smile:?

My thinking was that if Cloudformation doesn't require these to be specified when `healthyRetryPolicy` is specified, we can omit them here and have Cloudformation figure the defaults for us?